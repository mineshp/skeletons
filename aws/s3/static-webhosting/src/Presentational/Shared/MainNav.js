import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import Menu from 'semantic-ui-react/dist/commonjs/collections/Menu';
import logo from '../../logo.svg';

export default class MainNav extends Component {
  constructor(props) {
    super(props);
    this.state = {
      activeItem: 'view'
    };
    this.handleItemClick = this.handleItemClick.bind(this);
  }

  handleItemClick(e, { name }) {
    this.setState({ activeItem: name });
  }

  render() {
    const { activeItem } = this.state;

    return (
      <div className="header-main">
        <Menu inverted>
          <Menu.Item
            as={Link}
            to="/"
            name="home"
            active={activeItem === 'home'}
            onClick={this.handleItemClick}
          >
            <img alt="logo" src={logo} />
          </Menu.Item>
        </Menu>
      </div>
    );
  }
}
