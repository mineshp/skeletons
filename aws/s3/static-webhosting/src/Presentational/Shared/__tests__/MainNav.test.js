import React from 'react';
import { shallow } from 'enzyme';
import MainNav from '../MainNav';

describe('Presentational > Home', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <MainNav />
    );
  });
  it('Renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('sets activeName on handleClick', () => {
    wrapper.instance().handleItemClick({}, { name: 'test' });
    expect(wrapper.state().activeItem).toEqual('test');
  });
});
