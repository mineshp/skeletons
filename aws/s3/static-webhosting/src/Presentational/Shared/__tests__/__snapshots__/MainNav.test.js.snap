// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Presentational > Home Renders correctly 1`] = `
<div
  className="header-main"
>
  <Menu
    inverted={true}
  >
    <MenuItem
      active={false}
      as={[Function]}
      name="home"
      onClick={[Function]}
      to="/"
    >
      <img
        alt="logo"
        src="md-logo-green.png"
      />
    </MenuItem>
  </Menu>
</div>
`;
